### DISCUSSION TITLE
Ability to skip the "pending" state

### DISCUSSION THREAD

**Discussion by ak0nst** [grafana/grafana#51885] [Alerting]
I have metrics with a 10s interval, so I set the evaluation interval to 10s for the alert rule.
In case of lone out-of-range metric item, alert instance changes state to Pending and then to Normal.
I'm proposing a new rule option that will skip the pending state and fire once the threshold is exceeded.

**Comment by konrad147** [comment]
Hey, you can set the `for` value to 0, which effectively skips the pending state

**Reply by nikkodsg** [reply]
Do Mimir/Loki alerts support this? Right now, when I'm trying to set the evaluation interval with a value greater than the `For`, I'm getting this error: `Invalid evaluation interval. Evaluation interval should be smaller or equal to 'For' values for existing rules in this group.`

I want to run my alert evaluation every hour and immediately fire a notification.

**Reply by precisionconage** [reply]
I have the exact same use case as @nikkodsg. I'm able to hack around this by first setting `For` to a longer interval for each existing rule in the group, changing the group evaluation time to 1h, and then changing the rules' `For`s back to 0, but I'm not sure if this will cause problems/whether it's actually working as intended.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
